### PR TITLE
build-script for search_api_autocomplete:8.x-1.5

### PR DESCRIPTION
**Pull request by rajusah15**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

**Comment by seth-priya** [comment]
@prashantkhoje please review

File: d/drupal-search_api_autocomplete/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Then run a container using that image.
+
+    #docker run -it -d drupal_image /bin/bash
+    #docker ps
+    #docker exec -it 69cabf536f6f /bin/bash
+
+Live drupal webserver will be available at http://<ip>:8081
+
+After that we need to run below command inside container.Run it in single command . Step 1 will take sometime to exec.
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/README.md:40]:
```suggestion
After that we need to run below command inside the container. Run it in single command. Step 1 will take some time to exec.
```

File: d/drupal-search_api_autocomplete/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    #git clone https://git.drupalcode.org/project/search_api_autocomplete.git
+    #git checkout <versions>
+
+Clone the prerequisite dependency like ctools
+
+    #git clone https://git.drupalcode.org/project/ctools
+    #git checkout <versions>
+	#git clone https://git.drupalcode.org/project/search_api.git
+	#git checkout <versions>
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/README.md:65]:
```suggestion
Step 1:-
    # su - postgres -c 'if [[ $(psql -l | grep dru2_pg) ]]; then     echo "Database already configured..."; else    /usr/pgsql-13/bin/pg_ctl -D /var/lib/pgsql/13/data/ start;     /usr/pgsql-13/bin/createdb -T template0 dru2_pg;     psql dru2_pg < /opt/app-root/src/drupal_schema.sql; fi'

Step 2:-
    # /usr/sbin/httpd -k start;

Go to :-
    # cd /opt/app-root/src/drupal/modules

Clone the module which you wanted to test :-
    # git clone https://git.drupalcode.org/project/search_api_autocomplete.git
    # git checkout <versions>

Clone the prerequisite dependency like ctools
    # git clone https://git.drupalcode.org/project/ctools
    # git checkout <versions>
	# git clone https://git.drupalcode.org/project/search_api.git
	# git checkout <versions>
```

File: d/drupal-search_api_autocomplete/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    bash-4.4# ./vendor/bin/drush en contact_block
+    [success] Successfully enabled: contact_block
+
+
+    cd /opt/app-root/src/drupal/core
+    bash-4.4# pwd
+    /opt/app-root/src/drupal/core
+
+
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/README.md:93]:
```suggestion
Follow automate_drupal.sh for more detail:-
    # cd /opt/app-root/src/drupal
    # yum install -y git php php-gd

    # cd /opt/app-root/src/drupal/modules/contact_block
    # ./vendor/bin/drush en ctools
    [success] Successfully enabled: ctools
	
	# ./vendor/bin/drush en search_api
	[success] Successfully enabled: search_api

    # ./vendor/bin/drush en contact_block
    [success] Successfully enabled: contact_block

    # cd /opt/app-root/src/drupal/core

```

File: d/drupal-search_api_autocomplete/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+----------------
+PHPUnit 7.5.20 by Sebastian Bergmann and contributors.
+
+
+
+Time: 223 ms, Memory: 4.00 MB
+
+No tests executed!
+
+
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/README.md:110]:
```suggestion
RUN TEST:-
----------
    #  ../vendor/phpunit/phpunit/phpunit ../modules/search_api_autocomplete/src/Tests/

Test output
----------------
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.
Time: 223 ms, Memory: 4.00 MB

No tests executed!
```

File: d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+# Variables
+PACKAGE_NAME=search_api_autocomplete
+CORE_PACKAGE_NAME=drupal
+PACKAGE_URL=https://git.drupalcode.org/project/search_api_autocomplete.git
+CORE_PACKAGE_URL=https://github.com/drupal/drupal
+
+PACKAGE_VERSION=${1:-8.x-1.5}
+
+
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh:29]:
```suggestion
```

File: d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+composer require  drupal/jquery_ui
+#composer require  drush/drush
+
+if ! composer install --no-interaction; then
+        echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
+        exit 1
+fi
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh:58]:
```suggestion
# Check if package exists
if [ -d "$CORE_PACKAGE_NAME" ] ; then
    rm -rf $CORE_PACKAGE_NAME
    echo "$CORE_PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package"
fi

if ! git clone $CORE_PACKAGE_URL $CORE_PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$CORE_PACKAGE_URL $CORE_PACKAGE_NAME"
    echo "$CORE_PACKAGE_NAME  |  $CORE_PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi

cd $CORE_PACKAGE_NAME

composer require drupal/jquery_ui

if ! composer install --no-interaction; then
    echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
    exit 1
fi
```

File: d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+	cd search_api/
+	git checkout 8.x-1.7
+	cd ..
+
+ if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh:73]:
```suggestion
cd modules/
git clone https://git.drupalcode.org/project/ctools
	
git clone https://git.drupalcode.org/project/search_api.git
cd search_api/
git checkout 8.x-1.7
cd ..

if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi
```

File: d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails"
+        exit 1
+else
+        echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
+        exit 0
+fi
```
Comment (prashantkhoje) [inline d/drupal-search_api_autocomplete/drupal-search_api_autocomplete_ubi_8.5.sh:95]:
```suggestion
if ! ../vendor/phpunit/phpunit/phpunit ../modules/search_api_autocomplete/src/Tests/; then
    echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails"
    exit 1
else
    echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
    exit 0
fi
```

**Comment by rajusah15** [comment]
will raise PR in separate branch.

**Pull request closed without merging**